File: golang/791. 自定义字符串排序.go
func customSortString(order, s string) string {
    fre:=[26]int{}
    for _,c:=range s{
        fre[c-'a']++
    }
    t:=&strings.Builder{}
    for _,c:=range order{
        if fre[c-'a']>0{
            t.WriteString(strings.Repeat(string(c),fre[c-'a']))
            fre[c-'a']=0
        }
    }
    for i,k :=range fre{
        if k>0{
            t.WriteString(strings.Repeat(string('a'+i),k))
        }
    }
    return t.String()
}

File: 刷题/cpp/791. 自定义字符串排序.cpp
class Solution {
public:
    string customSortString(string order, string s) {
        vector<int> fre(26);
        for(char ch:s){
            fre[ch-'a']++;
        }
        string ans;
        for(char ch:order){
            if(fre[ch-'a']>0){
                ans+=string(fre[ch-'a'],ch);
                fre[ch-'a']=-0;
            }
        }
        for(int i=0;i<26;i++){
            if(fre[i]>0){
                ans+=string(fre[i],i+'a');
            }
        }
        return ans;
    }
};